x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformState.java

```diff
@@ -166,10 +146,7 @@ public String getWriteableName() {
     public void writeTo(StreamOutput out) throws IOException {
         taskState.writeTo(out);
         indexerState.writeTo(out);
-        out.writeBoolean(currentPosition != null);
-        if (currentPosition != null) {
-            out.writeMap(currentPosition);
-        }
+        out.writeMap(currentPosition);
```

Comment:
Will the map order be preserved? `StreamInput` has a private method `readLinkedHashMap` called by `readGenericValue`. It be best to use `readGenericValue` and `writeGenericValue`.

The unit tests won't fail because `DataFrameTransformState.equals` does will not consider map order. Maybe use LinkedHashMap in the declaration so it is explicit that insertion order should be preserved. Then if currentPosition  was a LinkedHashMap I would expect the tests to fail occasionally

Reply:
@davidkyle good call.

Reply:
Just verified that `parser.mapOrdered` returns a `LinkedHashMap`. But, somebody could pass the position manually the constructor. Will fix that.